publish-app-do/index.md

```diff
@@ -0,0 +1,414 @@
+
+```bash
+adduser appuser
+```
+
+You will be prompted to enter a new password for appuser. After entering a password, you will be asked for Full Name, Room Number, Work Phone, Home Phone or Others, hit ENTER to use the default values.
+
+The new user is created, to have it all the rights in the system you will need to add it to the `sudo` group.
+
+```bash
+usermod -aG sudo appuser
+```
+
+To be able to login as the new user, copy your ssh public key to the new users home directory
+
+```bash
+rsync --archive --chown=appuser:appuser ~/.ssh /home/appuser
+```
+
+To set up your server to install and run the Starport app, we will need to install a few tools which will help us to publish your blockchain app.
+
+Install [Go](https://golang.org/dl/).
+
+At the time of writing the version is 1.16.5.
+
+```bash
+cd ~
+wget https://golang.org/dl/go1.16.5.linux-amd64.tar.gz
+sha256sum go1.16.5.linux-amd64.tar.gz
+# b12c23023b68de22f74c0524f10b753e7b08b1504cb7e417eccebdd3fae49061  go1.16.5.linux-amd64.tar.gz
+```
+
+Extract the archive and keep it in /usr/local
+
+```bash
+sudo tar -xvf go1.16.5.linux-amd64.tar.gz -C /usr/local
+```
+
+You will now have a directory called go in the `/usr/local` directory. Next, recursively change this directory’s owner and group to `appuser`:
+
+```bash
+sudo chown -R appuser:appuser /usr/local/go
+```
+
+Add the go paths to your profile
+
+```bash
+nano ~/.profile
+```
+
+At the end of the file, add the following two lines
+
+```bash
+export GOPATH=$HOME/go
+export PATH=$PATH:$GOPATH/bin:/usr/local/go/bin
+```
+
+To update your profile
+
+```bash
+source ~/.profile
+```
+
+Now you can verify your go installation with 
+
+```bash
+go version
+```
+
+which should print
+
+```bash
+go version go1.16.5 linux/amd64
+```
+
+To use Starport, install the Starport CLI.
+
+```bash
+curl https://get.starport.network/starport! | sudo bash
+``` 
+
+## Upload your app to the Droplet
+
+There are a few methods available to upload your app to the Droplet. One convinient way might be to upload the app to GitHub and then download it on your Droplet again. Starport creates a new GitHub repository for you to make it convinient to add it to your repositories.
+
+We will upload the files now with `scp`, to avoid a few extra steps. 
+
+On your local machine, use the following command to upload the app directory to your server.
+
+`scp -r <appname> appuser@<IP address>:/home/appuser/`
+
+This uploads your local app to your Digital Ocean droplet, into the `/home/appuser` directory, the home directory of the `appuser`.
+
+After the upload finished, navigate on your Droplet into the `/home/appuser/<appname>` directory.
+
+## Install your app
+
+To install your app and configuration, use the Starport CLI
+
+```bash
+cd /home/appuser/<appname>
+starport serve
+```
+
+This will build your Starport app, create a genesis file and your configuration.
+You will see an output similar to
+
```

Comment:
This happens on first run:
```
cannot build app:

	error while running command /tmp/tsc4158630: : context canceled
Waiting for a fix before retrying...
```

Reply:
@fadeev 👆 FYI: this is an error that occurs when I run `starport serve` on an Ubuntu machine the first time. Aborting and rerunning the process fixes this.

Reply:
I didn't have this experience...